pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/PatchRoute.java

```diff
@@ -65,11 +63,7 @@ public Object processRequest(Request request, Response response, String userId)
                 throw new AuthorizationException("User is not authorized to patch participant data");
             }
         } catch (JsonSyntaxException e) {
-            throw new DSMBadRequestException("Invalid request payload format for patch");
-        } catch (DuplicateException e) {
-            // TODO: fix this by inspecting the DuplicateException throwers to see if cases are due to
-            // bad request data (400)
-            throw new DsmInternalError("Duplicate value", e);
+            throw new DSMBadRequestException("Invalid request payload format for patch: " + requestBody, e);
```

Comment:
Let me know if anyone is worried about logging a potentially very large requestBody.  I think we want to log this for a while to help us determine what kind of posts should result in a 400 vs. 500.

Reply:
The request bodies I've seen are manageable, so I do not think logging it is an issue.
If the body cannot be mapped to `Patch.class`, that is a 400, so this code is correct as DSMBadRequestException (before and after this change). But I may be missing the intent of your comment.